### PR TITLE
Adds ordering, filtering, and searching capabilities to the API 

### PR DESCRIPTION
**Pull request by abrahammartin**
New API point created with ordering, filtering, and searching capabilities. Documentation at: http://localhost:8080/docs/#assets-advanced-list-list

### CODE REVIEW

File: assets/urls.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 )
 
 urlpatterns = [
+    path('assets/advanced-list/', AssetAdvanceList.as_view(), name="assets-advanced-list"),
```
Comment (rjw57) [inline assets/urls.py]: Maybe just have this be the ``assets/`` endpoint?

**Comment by rjw57** [comment]
I think we should also add db_index=True to all the fields we can sort/filter by. Otherwise performance is going to suck.

File: assets/urls.py (hunk 1 above)
Comment (abrahammartin) [reply assets/urls.py]: that doesn't work because the POST method (create a new asset) also uses assets/

**Comment by abrahammartin** [comment]
good point

File: assets/urls.py (hunk 1 above)
Comment (rjw57) [reply assets/urls.py]: Why does advance-list require POST? It's surely a GET-only request?

Comment (rjw57) [reply assets/urls.py]: Ah, I think I see - I mean, why not have GET assets/ point to AssetAdvanceList.list() but keep POST as it is?

**Comment by rjw57** [comment]
Tests are failing and I still think the sort/filter functionality should be on the GET assets/ endpoint.

**Pull request merged**